packages/react-native/src/primitives/Icon/styles.ts

```diff
@@ -1,16 +1,22 @@
 import { StyleSheet } from 'react-native';
 
+import { StrictTheme } from '../../theme';
 import { IconProps, IconStyles } from './types';
 
-export const getStyles = (
-  color: IconProps['color'],
-  size: IconProps['size']
-): IconStyles =>
-  StyleSheet.create({
+export const getThemedStyles = (
+  theme: StrictTheme,
+  color?: IconProps['color'],
+  size?: IconProps['size']
+): IconStyles => {
+  const { components } = theme.tokens;
+
+  return StyleSheet.create({
     icon: {
       height: size,
```

Comment:
What happens if the `size` is `undefined`?

Reply:
That's a good call out. Should we add the same default size we use for the `IconButton` in the `Icon` and set `size` as required on line 9? Might make sense to do something for `color` as well. What do you think @ioanabrooks

Reply:
Yup! Added tokens for icon sizes, the default value is medium coming through the default theme in case size is undefined. For color we should use one of the tokens, otherwise RN will use black if it's undefined, not great for dark mode. Not sure how we are supporting dark mode or which token I should use atm.